core/src/test/java/edu/wpi/grip/core/operations/network/http/HttpPublisherTest.java

```diff
@@ -0,0 +1,183 @@
+import edu.wpi.grip.core.operations.network.NumberPublishable;
+import edu.wpi.grip.core.settings.ProjectSettings;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.InputSocketImpl;
+import edu.wpi.grip.core.sockets.SocketHint;
+import edu.wpi.grip.util.GRIPCoreTestModule;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.BasicHttpEntity;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class HttpPublisherTest {
+
+    private static final String dataPath = "/GRIP/data";
+    private static final String noDataPath = "/GRIP/data?no_data_here";
+    private static final String empty = "{}";
+    private static final String name = "foo";
+    private static final String json = "{\n  \"foo\": 1.0\n}";
+    private static final String unexpectedResponseMsg = "Unexpected response to data request";
+
+    private EventBus eventBus;
+    private GripServer server;
+    private DataHandler dataHandler;
+    private HttpPublishOperation<Number, NumberPublishable> operation;
+
+    private HttpClient client;
+
+    @Before
+    public void setUp() {
+        GRIPCoreTestModule m = new GRIPCoreTestModule();
+        m.setUp();
+        Injector injector = Guice.createInjector(m);
+        InputSocket.Factory isf = injector.getInstance(InputSocket.Factory.class);
+        m.tearDown();
+        eventBus = new EventBus();
+        server = GripServerTest.makeServer(new GripServerTest.TestServerFactory(), ProjectSettings::new);
+        unclaimDataHandler();
+        dataHandler = new DataHandler(eventBus);
+
+        operation = new HttpPublishOperation<>(
+                isf,
+                Number.class,
+                NumberPublishable.class,
+                NumberPublishable::new,
+                new HttpPublishManager(server, dataHandler)
+        );
+
+        client = HttpClients.createDefault();
+
+        server.addHandler(dataHandler);
+        server.start();
+    }
+
+    private void unclaimDataHandler() {
+        try {
+            Field f = GenericHandler.class.getDeclaredField("claimedContexts");
+            f.setAccessible(true);
+            @SuppressWarnings("unchecked")
+            Set<String> claimedContexts = (Set<String>) f.get(null);
+            claimedContexts.remove(GripServer.DATA_PATH);
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+            fail("Could not unclaim /GRIP/data");
```

Comment:
You're suppressing the stack trace here.
I'd throw an `AssertionError` instead and pass the exception into that.